lib/new_relic/agent/agent.rb

```diff
@@ -788,6 +767,34 @@ def graceful_disconnect
         end
       end
 
+      def shutdown_message
+        if @connected_pid == $$ && !@service.kind_of?(NewRelic::Agent::NewRelicService)
+          ::NewRelic::Agent.logger.debug("Sending New Relic service agent run shutdown message")
+          @service.shutdown
```

Comment:
I remember struggling with the name for this one. I'm still not certain if `shutdown_message` is quite right since `@service.shutdown` is called, which actually performs the shutdown operation. We could call it `shutdown_service` or `service_shutdown` since that's the goal, with logging messages also included. What are your thoughts?

Reply:
yeah that makes sense. I like `shutdown_service` personally.